pkg/operator/starter.go

```diff
@@ -281,6 +282,24 @@ func RunOperator(ctx context.Context, controllerContext *controllercmd.Controlle
 		controllerContext.EventRecorder,
 	)
 
+	// create the two node with fencing (TNF) operator deployment on DualReplica clusters
+	var tnfDeploymentReconciler factory.Controller
+	if cpTopology, err := ceohelpers.GetControlPlaneTopology(configInformers.Config().V1().Infrastructures().Lister()); err != nil {
+		return err
+	} else if configv1.DualReplicaTopologyMode == cpTopology {
+		tnfDeploymentReconciler = tnfdeploymentcontroller.NewTnfDeploymentController(
+			ctx,
+			operatorClient,
+			kubeClient,
+			kubeInformersForNamespaces.InformersFor("openshift-etcd"),
+			kubeInformersForNamespaces,
+			configInformers.Config().V1().Infrastructures(),
+			envVarController,
+			controllerContext.EventRecorder,
+			os.Getenv("IMAGE"),
```

Comment:
couldn't you call `tnfDeploymentReconciler.Run(...)` here, instead than guarding the call later with a ` if tnfDeploymentReconciler != nil` ?

Reply:
I said that because a bunch of other controllers do it similarly (e.g. https://github.com/openshift/cluster-etcd-operator/blob/ba5c2e88e7c489978fa847aee8afc8fb99a30e65/pkg/operator/starter.go#L522-L524)